src/workflows/registration.ts

```diff
@@ -51,12 +51,11 @@ export async function registerOffchainWorkflow(
 }
 
 export async function registerOffchainWorkflowWithSigner(
-  l1Signer: Signer,
-  l2Signer: L2Signer,
+  walletConnection: WalletConnection,
   usersApi: UsersApi,
 ): Promise<void> {
-  const userAddress = await l1Signer.getAddress();
-  const starkPublicKey = l2Signer.getAddress();
+  const userAddress = await walletConnection.l1Signer.getAddress();
+  const starkPublicKey = walletConnection.l2Signer.getAddress();
```

Comment:
I suppose that we should have await here as well. As we cannot guarantee that another l2 signer can get address without Promise

Reply:
If I understood it correctly, I'd say you're right, but since we don't need it for the moment, what do you think of using it according to our needing right now and addressing this concern when the need comes in?

Reply:
It's better to fix in the beginning, as it's a strange user experience to have different interfaces for L1 and L2.

Reply:
@carmen0208 @yundi-fu @vmiguez-immutable I agree with her on this topic. If you don't mind, I'll create a tech debt to do this refactoring and handle the consequent side effects (since this change is not exactly related to this task). Can you guys let us know if you disagree?

Reply:
created: https://immutable.atlassian.net/browse/WT-614